tchannel-core/src/main/java/com/uber/tchannel/tracing/OpenTracingContext.java

```diff
@@ -52,31 +52,31 @@ public OpenTracingContext(@NotNull ScopeManager scopeManager) {
     @Override
     @SuppressWarnings("resource")
     public void pushSpan(@NotNull Span span) {
-        scopeManager.activate(span, false);
+        scopeManager.activate(span);
     }
 
     @Override
     @SuppressWarnings("resource")
     public boolean hasSpan() {
-        return scopeManager.active() != null;
+        return scopeManager.activeSpan() != null;
     }
 
     @Override
     public @NotNull Span currentSpan() throws EmptyStackException {
-        @SuppressWarnings("resource") Scope scope = scopeManager.active();
-        if (scope == null) {
+        if (!hasSpan()) {
             throw new EmptyStackException();
         } else {
-            return scope.span();
+            return scopeManager.activeSpan();
         }
     }
 
     @Override
     public @NotNull Span popSpan() throws EmptyStackException {
-        try (Scope scope = scopeManager.active()) {
-            if (scope == null) {
-                throw new EmptyStackException();
-            } else {
+        if (!hasSpan()) {
+            throw new EmptyStackException();
+        } else {
+            //auto close scope acts as if we pop the span
+            try (Scope scope = scopeManager.active()) {
```

Comment:
@yurishkuro @vprithvi I noticed that opentracing has marked scopeManager.active() as deprecated, they provide activeSpan() to access the span. However here we need the access to scope so that when the current span is popped, the scope can be closed and thus threadLocal can set the saved scope as active. I am afraid if opentracing remove the act() method, it will break this OpentracingContext, that means we may also need to deprecate this custom context implemented provided by tchannel and ask tchannel users to use scope manager instead, which may be another big task.

Reply:
OpenTracing API my be  frozen soon in favor  of OpenTelemetry. But yet, the right thing would be to change tchannel implementation to use scopes. I just don't think it's pressing.